File: year-2/programming-design/c#/motorcycle-contest/MotorcycleContest/Domain/Dtos/ParticipantInfo.cs
﻿using System;

using MotorcycleContest.Domain.Entities;

namespace MotorcycleContest.Domain.Dtos
{
    public class ParticipantInfo
    {
        public Race Race { get; set; }
        public Participant Participant { get; set; }

        public ParticipantInfo(Race race, Participant participant)
        {
            this.Race = race;
            this.Participant = participant;
        }

        public override string ToString()
        {
            return "Name: " + Participant.Name + ", Race: " + Race.Capacity;
        }
    }
}

File: year2/mpp/java/MotorcycleContest/src/main/java/domain/dtos/ParticipantInfo.java
package domain.dtos;

import domain.entities.Participant;
import domain.entities.Race;

public class ParticipantInfo {
    private Race race;
    private Participant participant;

    public ParticipantInfo(Race race, Participant participant) {
        this.race = race;
        this.participant = participant;
    }

    public Race getRace() {
        return race;
    }

    public void setRace(Race race) {
        this.race = race;
    }

    public Participant getParticipant() {
        return participant;
    }

    public void setParticipant(Participant participant) {
        this.participant = participant;
    }

    @Override
    public String toString() {
        return "Name: " + participant.getName() + ", Race: " + race.getCapacity() + "cc";
    }
}
